src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newAction = [];
+  let stateClones = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destructuring for `actions`
```suggestion
  for (const action of actions) {
    const { type, ........ } = action;
    
    switch (type) {
```